### PR TITLE
Add documentation of Unmanaged Bootloader builds

### PR DESCRIPTION
**Pull request by theotherjimmy**
Documentation for https://github.com/ARMmbed/mbed-os/pull/4202

# DO NOT MERGE BEFORE https://github.com/ARMmbed/mbed-os/pull/4202 IS MERGED

### CODE REVIEW

File: docs/advanced/bootloader.md [hunk 1]
```diff
@@ -6,7 +6,11 @@ This guide explains how to create a bootloader, how to create a main program to
 
 A target requires flash HAL functionality in order to support the bootloader. Please follow [the bootloader porting guide](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/flash/) for more details.
 
-## Creating the bootloader
+## mbed OS managed bootloader
+
+The tools of mbed OS know how to manage some bootloader projects. The tools can manage bootloader projects where the bootlodoader comes before the application in ROM and the application starts immediately after the bootloader. If your bootloader does not meet both of these requirements, then please read the [unmanaged bootloader section](bootloader.md#unmanaged-bootloader). A managed bootloader project automatically merge the bootloader image with the application image as part of the application image build process.
```
Comment (AnotherButler) [inline docs/advanced/bootloader.md]: Please change "automatically merge" to "automatically merges".

File: docs/advanced/bootloader.md [hunk 2]
```diff
@@ -94,3 +98,18 @@ It produces the following ROM layout:
 ``` 
 
 For an example showing how to create an application that uses a bootloader, see the [mbed OS bootloader example](https://github.com/armmbed/mbed-os-example-bootloader-blinky) repository.
+
+## Unmanaged bootloader
+
+You want to have an unmanaged bootloader when your bootloader's requirements conflict with the requirements of the managed bootloader. You need an unmanaged bootloader when your bootloader does not come before your application in ROM or your application does not start immediately after your bootloader. Unlike a managed bootloader, an unmananged bootloader does not automatically merge the bootloader image with the application image after building the application. We expect users of an unmanaged bootloader build to construct there own set of scripts built atop the `mbed compile` primitive to perform bootloader and application merging.
```
Comment (AnotherButler) [inline docs/advanced/bootloader.md]: Please change "there own set" to "their own set".

File: docs/advanced/bootloader.md [hunk 3]
```diff
@@ -94,3 +98,18 @@ It produces the following ROM layout:
...
+
+An unmanaged bootloader build is a method for controlling the link location of a program within mbed OS. There are two configuration options available for changing the link location: `target.mbed_app_start` and `target.mbed_app_size`. 
+
+### `target.mbed_app_start`
+
+
```
Comment (AnotherButler) [inline docs/advanced/bootloader.md]: Please delete extra line.

File: docs/advanced/bootloader.md [hunk 4]
```diff
@@ -94,3 +98,18 @@ It produces the following ROM layout:
...
+The configuration option `target.mbed_app_start` sets the starting address of the linker script by defining the `MBED_APP_START` macro for the linker script. You may only define this configuration option within the `target_overrides` section of an mbed application configuration, and you may not define it for the meta-target `*`. When you do not define this configuration option, it defaults to the start of a target's ROM. This configuration option must be an address within ROM.
```
Comment (AnotherButler) [inline docs/advanced/bootloader.md]: Please delete the hyphen in "metatarget".

File: docs/advanced/bootloader.md [hunk 5]
```diff
@@ -94,3 +98,18 @@ It produces the following ROM layout:
...
+
+### `target.mbed_app_size`
+
+The configuration option `target.mbed_app_size` defines the size of an application image in ROM by defining the `MBED_APP_SIZE` macro for the linker script. You may only define this configuration option on a per-target basis defined within the `target_overrides` section of an mbed application configuration, and you may not define it for the meta-target `*`. When you do not define this configuration option, it defaults to the  remaining ROM, which the mbed OS tools calculate by subtracting the image's offset into ROM from the total size of ROM. Together with `target.mbed_app_start`, these configuration options define a continuous region of memory that an image may use. The tools verify that this region of memory is in ROM, but the tools do not perform any other checks for consistency or validity.
```
Comment (AnotherButler) [inline docs/advanced/bootloader.md]: - Please delete the hyphen in "metatarget".
- Please delete the extra space between "to the" and "remaining ROM".
- Consider splitting the third sentence into two shorter sentences.

**Comment by theotherjimmy** [comment]
@AnotherButler Fixed

**Comment by theotherjimmy** [comment]
@AnotherButler The dependent PR on mbed-os was merged. LGTM!

**Pull request merged**